tools/plugins/com.liferay.ide.project.core/src/com/liferay/ide/project/core/util/ProjectUtil.java

```diff
@@ -1338,6 +1338,19 @@ public static boolean isLiferaySDKProjectDir(File file) {
 		return false;
 	}
 
+	public static boolean isGradleProject(IProject project) {
+		boolean retval = false;
+
+		try {
+			retval = FileUtil.exists(project) && project.hasNature("org.eclipse.buildship.core.gradleprojectnature") &&
+				FileUtil.exists(project.getFile("build.gradle"));
```

Comment:
not every gradle project has build.gradle. Maybe just checking nature is enough?